easybuild/easyblocks/o/openmm.py

```diff
@@ -0,0 +1,132 @@
+##
+# Copyright 2009-2015 Ghent University
+# Copyright 2015-2016 Stanford University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing OpenMM, implemented as an easyblock
+
+@author: Stephane Thiell (Stanford University)
+"""
+import os
+from distutils.version import LooseVersion
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_OpenMM(CMakeMake):
+    """Support for building/installing OpenMM."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialisation of custom class variables for OpenMM."""
+        super(EB_OpenMM, self).__init__(*args, **kwargs)
+
+        self.example = None
+
+    def configure_step(self):
+        """Custom configuration procedure for OpenMM."""
+
+        if LooseVersion(self.version) < LooseVersion('6.3'):
+            self.log.warn("OpenMM easyblocks not tested for older version of OpenMM")
+
+        # build a release build
+        self.cfg.update('configopts', "-DCMAKE_BUILD_TYPE=Release")
+
+        #self.cfg.update('configopts', "-DOPENMM_BUILD_SHARED_LIB=OFF")
+        #self.cfg.update('configopts', "-DOPENMM_BUILD_STATIC_LIB=ON")
+
+        # CUDA is required
+        cuda = get_software_root('CUDA')
+        if cuda:
+            self.cfg.update('configopts', "-DCUDA_TOOLKIT_ROOT_DIR=%s" % cuda)
+        else:
+            raise EasyBuildError("CUDA not found")
+
+        doxygen = get_software_root('Doxygen')
+        if not doxygen:
+            raise EasyBuildError("Doxygen not found")
+
+        doxygen = get_software_root('SWIG')
+        if not doxygen:
+            raise EasyBuildError("SWIG not found")
```

Comment:
this can be simplified a bit by using a for loop over the 3 required dependencies